client/src/views/NewPlace/index.js

```diff
@@ -29,6 +29,12 @@ class NewPlaceClass extends React.Component {
     this.props.getLists();
   }
 
+  componentDidUpdate(prevProps) {
```

Comment:
What does this do? The logic doesn't look right to me. Why would a view requires a componentDidUpdate where it compares its props?

Reply:
When I make a new place, I need to call createPlace action and then addPlaceToList action. But I need the placeId to addPlaceToList, and I only get the new placeId asynchronously after I call createPlace.

So I first call createPlace, then after it's done it updates props with the new placeId (which is previously null), then when the placeId is available call addPlaceToList.

Originally, the createPlace action had two API calls - create the place, then add place to list. But then I thought it wasn't good for one action function to do two things, making it not pure I think? So I changed it to this. 
Sorry it's confusing, what's the best way to do this?

Reply:
Hmm, I am not sure. I'm not really following just by a glance in the code. Let's just keep it this way.